source/Locales/Compactness/Properties.lagda

```diff
@@ -0,0 +1,716 @@
+open import UF.Logic
+open import UF.Powerset-Fin pt hiding (⟨_⟩)
+open import UF.Powerset-MultiUniverse
+open import UF.Sets-Properties
+
+open AllCombinators pt fe
+open Locale
+open PropositionalTruncation pt
+
+\end{code}
+
+\section{Preliminaries}
+
+Given a family `S`, we denote the type of its subfamilies by `SubFam S`.
+
+\begin{code}
+
+SubFam : {A : 𝓤  ̇} {𝓦 : Universe} → Fam 𝓦 A → 𝓦 ⁺  ̇
+SubFam {𝓤} {A} {𝓦} (I , α) = Σ J ꞉ 𝓦  ̇ , (J → I)
+
+\end{code}
+
+Tiny lemma recording the fact that nothing is a member of the empty list.
+
+\begin{code}
+
+not-in-empty-list : {A : 𝓤  ̇} {x : A} → ¬ ∥ member x [] ∥
+not-in-empty-list = ∥∥-rec 𝟘-is-prop (λ ())
+
+\end{code}
+
+Given any list, the type of elements that fall in the list is a
+Kuratowski-finite type.
+
+\begin{code}
+
+nth : {X : 𝓤  ̇} → (xs : List X) → (i : Fin (length xs)) → Σ x ꞉ X , ∥ member x xs ∥
+nth         (x ∷ _)  (inr ⋆) = x , ∣ in-head ∣
+nth {_} {X} (_ ∷ xs) (inl n) = x , ∥∥-functor in-tail (pr₂ IH)
+ where
+  IH : Σ x ꞉ X , ∥ member x xs ∥
+  IH = nth xs n
+
+  x : X
+  x = pr₁ IH
+
+nth-is-surjection : {X : 𝓤  ̇} (xs : List X) → is-surjection (nth xs)
+nth-is-surjection []       (y , μ) = ∥∥-rec ∃-is-prop (λ ()) μ
+nth-is-surjection (x ∷ xs) (y , μ) = ∥∥-rec ∃-is-prop † μ
+ where
+  † : member y (x ∷ xs) → ∃ i ꞉ Fin (length (x ∷ xs)) , (nth (x ∷ xs) i ＝ y , μ)
+  † in-head     = ∣ inr ⋆ , to-subtype-＝ (λ _ → ∥∥-is-prop) refl ∣
+  † (in-tail p) = ∥∥-rec ∃-is-prop ‡ IH
+   where
+    IH : (y , ∣ p ∣) ∈image nth xs
+    IH = nth-is-surjection xs (y , ∣ p ∣)
+
+    ‡ : Σ i ꞉ Fin (length xs) , (nth xs i ＝ y , ∣ p ∣)
+      → ∃ i ꞉ Fin (length (x ∷ xs)) , (nth (x ∷ xs) i ＝ y , μ)
+    ‡ (i , q) = ∣ inl i , to-subtype-＝ (λ _ → ∥∥-is-prop) (pr₁ (from-Σ-＝ q)) ∣
+
+list-members-is-Kuratowski-finite : {X : 𝓤  ̇}
+                                  → (xs : List X)
+                                  → is-Kuratowski-finite (Σ x ꞉ X , ∥ member x xs ∥)
+list-members-is-Kuratowski-finite {𝓤} {A} xs =
+ ∣ length xs , nth xs , nth-is-surjection xs ∣
+
+\end{code}
+
+TODO: The function `nth` above should be placed in a more appropriate module.
+
+\section{Alternative definition of compactness}
+
+Compactness could have been alternatively defined as:
+
+\begin{code}
+
+is-compact-open' : (X : Locale 𝓤 𝓥 𝓦) → ⟨ 𝒪 X ⟩ → Ω (𝓤 ⊔ 𝓥 ⊔ 𝓦 ⁺)
+is-compact-open' {𝓤} {𝓥} {𝓦} X U =
+ Ɐ S ꞉ Fam 𝓦 ⟨ 𝒪 X ⟩ ,
+  U ≤ (⋁[ 𝒪 X ] S) ⇒
+   (Ǝ (J , h) ꞉ SubFam S ,
+       is-Kuratowski-finite J
+     × (U ≤ (⋁[ 𝒪 X ] ⁅  S [ h j ] ∣ j ∶ J ⁆)) holds)
+  where
+   open PosetNotation (poset-of (𝒪 X))
+
+\end{code}
+
+This is much closer to the “every cover has a finite subcover” definition from
+point-set topology.
+
+It is easy to show that this implies the standdard definition of compactness,
+but we need a bit of preparation first.
+
+Given a family `S`, we denote by `family-of-lists S` the family of families
+of lists of `S`.
+
+\begin{code}
+
+module Some-Lemmas-On-Directification (F : Frame 𝓤 𝓥 𝓦) where
```

Comment:
Why not everything in lower case? Either use hyphens (preferred) or camel case.

Reply:
We should also document this convention in the contribution guidelines. I was under the impression that we used capital kebab casing for module names. Do you want me add it @martinescardo? Is there anything else you would like to add?